tests/component/test_stream_readers.py

```diff
@@ -78,3 +171,384 @@ def test___analog_multi_channel_reader___read_many_sample_with_wrong_dtype___rai
+    return _pwr_current_to_codes(_get_current_setpoint_for_chan(chan_index))
+
+
+@pytest.fixture
+def pwr_single_channel_task(
+    task: nidaqmx.Task, sim_ts_power_device: nidaqmx.system.Device
+) -> nidaqmx.Task:
+    task.ai_channels.add_ai_power_chan(
+        f"{sim_ts_power_device.name}/power",
+        _get_voltage_setpoint_for_chan(0),
+        _get_current_setpoint_for_chan(0),
+        True,  # output enable
+    )
+    return task
+
+
+@pytest.fixture
+def pwr_multi_channel_task(
+    task: nidaqmx.Task, sim_ts_power_devices: List[nidaqmx.system.Device]
+) -> nidaqmx.Task:
+    for chan_index, sim_ts_power_device in enumerate(sim_ts_power_devices):
+        task.ai_channels.add_ai_power_chan(
+            f"{sim_ts_power_device.name}/power",
+            _get_voltage_setpoint_for_chan(chan_index),
+            _get_current_setpoint_for_chan(chan_index),
+            True,  # output enable
+        )
+    return task
+
+
+def test___power_single_channel_reader___read_one_sample___returns_valid_samples(
+    pwr_single_channel_task: nidaqmx.Task,
+) -> None:
+    reader = PowerSingleChannelReader(pwr_single_channel_task.in_stream)
+
+    data = reader.read_one_sample()
+
+    assert data.voltage == pytest.approx(_get_voltage_setpoint_for_chan(0), abs=POWER_EPSILON)
+    assert data.current == pytest.approx(_get_current_setpoint_for_chan(0), abs=POWER_EPSILON)
+
+
+def test___power_single_channel_reader___read_many_sample___returns_valid_samples(
+    pwr_single_channel_task: nidaqmx.Task,
+) -> None:
+    reader = PowerSingleChannelReader(pwr_single_channel_task.in_stream)
+    samples_to_read = 10
+    voltage_data = numpy.full(samples_to_read, math.inf, dtype=numpy.float64)
+    current_data = numpy.full(samples_to_read, math.inf, dtype=numpy.float64)
+
+    samples_read = reader.read_many_sample(voltage_data, current_data, samples_to_read)
+
+    assert samples_read == samples_to_read
+    assert voltage_data == pytest.approx(_get_voltage_setpoint_for_chan(0), abs=POWER_EPSILON)
+    assert current_data == pytest.approx(_get_current_setpoint_for_chan(0), abs=POWER_EPSILON)
+
+
+def test___power_single_channel_reader___read_many_sample_with_wrong_dtype___raises_error_with_correct_dtype(
+    pwr_single_channel_task: nidaqmx.Task,
+) -> None:
+    reader = PowerSingleChannelReader(pwr_single_channel_task.in_stream)
+    samples_to_read = 10
+    invalid_data = numpy.full(samples_to_read, math.inf, dtype=numpy.float32)
+    valid_data = numpy.full(samples_to_read, math.inf, dtype=numpy.float64)
+
+    with pytest.raises((ctypes.ArgumentError, TypeError)) as exc_info:
+        _ = reader.read_many_sample(valid_data, invalid_data, samples_to_read)
+
+    assert "float64" in exc_info.value.args[0]
+
+    with pytest.raises((ctypes.ArgumentError, TypeError)) as exc_info:
+        _ = reader.read_many_sample(invalid_data, valid_data, samples_to_read)
+
+    assert "float64" in exc_info.value.args[0]
```

Comment:
These power error tests have too many act/assert clauses. How about parameterizing them on voltage_dtype and current_dtype so you can test both cases with one act clause and one assert clause?